New resource `azurerm_virtual_machine_gallery_application_assignment` (#22945)
hashicorp/terraform-provider-azurerm: website/docs/r/virtual_machine_gallery_application_assignment.html.markdown

---
subcategory: "Compute"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_virtual_machine_gallery_application_assignment"
description: |-
  Manages a Virtual Machine Gallery Application Assignment.
---

# azurerm_virtual_machine_gallery_application_assignment

Manages a Virtual Machine Gallery Application Assignment.

> ~> **NOTE:** Gallery Application Assignment can be defined either directly on `azurerm_linux_virtual_machine` and `azurerm_windows_virtual_machine` resources, or using the `azurerm_virtual_machine_gallery_application_assignment` resource - but the two ways cannot be used together. If both are used against the same Virtual Machine, spurious changes will occur.

Reviewer: This note should also suggest the use of `ignore_changes` on the virtual machine resource for the `gallery_application` block.

```suggestion
~> **Note** Gallery Application Assignments can be defined either directly on `azurerm_linux_virtual_machine` and `azurerm_windows_virtual_machine` resources, or using the `azurerm_virtual_machine_gallery_application_assignment` resource - but the two approaches cannot be used together. If both are used with the same Virtual Machine, spurious changes will occur. It's recommended to use `ignore_changes` for the `gallery_application` block on the associated virtual machine resources, to avoid a persistent diff when using this resource.
```

  container_access_type = "blob"
}

resource "azurerm_storage_blob" "example" {
  name                   = "scripts"
  storage_account_name   = azurerm_storage_account.example.name
  storage_container_name = azurerm_storage_container.example.name
  type                   = "Block"
  source_content         = "[scripts file content]"
}

resource "azurerm_gallery_application_version" "example" {
  name                   = "0.0.1"
  gallery_application_id = azurerm_gallery_application.example.id
  location               = azurerm_gallery_application.example.location

  manage_action {
    install = "[install command]"
    remove  = "[remove command]"
  }

  source {
    media_link = azurerm_storage_blob.example.id
  }

  target_region {
    name                   = azurerm_gallery_application.example.location
    regional_replica_count = 1
  }
}

resource "azurerm_virtual_machine_gallery_application_assignment" "example" {
  gallery_application_version_id = azurerm_gallery_application_version.example.id
  virtual_machine_id             = data.azurerm_virtual_machine.example.id
}
```

## Arguments Reference

The following arguments are supported:

> * `gallery_application_version_id` - (Required) The ID of the Gallery Application Version. Changing this forces a new resourec to be created.

Reviewer: Spelling

```suggestion
* `gallery_application_version_id` - (Required) The ID of the Gallery Application Version. Changing this forces a new resource to be created.
```

> * `virtual_machine_id` - (Required) The ID of the Virtual Machine. Changing this forces a new resourec to be created.

Reviewer: ```suggestion
* `virtual_machine_id` - (Required) The ID of the Virtual Machine. Changing this forces a new resource to be created.
```
